pyproject.toml

```diff
@@ -22,6 +22,33 @@ extend_skip_glob = ["**/migrations/*"]
 
 ##############################################
 
+[tool.ruff]
+target-version = "py310"
+line-length = 120
+select = ["F", "E", "B", "W", "N", "UP", "YTT", "FIX", "ASYNC", "A", "DJ", "EXE", "ICN", "G", "SLOT", "TID", "TCH", "INT", "C4", "ISC", "INP", "PIE", "RSE", "RET", "COM","PGH","FLY", "PERF", "PLE"]  # PYI,Q,SIM,PTH,PL,TRY,RUF
+ignore = [
+    "E501",                  # line-too-long: black does code formatting for us
+    "FIX004",                # hacks should be possible
+    "A003",                  # we shadow some builtins sometime
+    "DJ008",                 # do not use __str__ where it is not required
+    "COM812",                # incompatible with formatter
+    "PERF203",
+]
+
+ignore-init-module-imports = true
+pep8-naming.extend-ignore-names = ["assert*", "*Formset"]  # custom assert methods use camelCase; Formsets use PascalCase
+
+[tool.ruff.per-file-ignores]
+"**/migrations/*.py" = ["N806"]  # migrations have model classes as local variables, we use PascalCase for these
+"**/commands/*.py" = ["T201"]  # printing in commands is ok
```

Comment:
Why does github render this with full-red background? :D

Reply:
seems to break here but is ok in the rest of the file